api_functional_v4_tests.go

```diff
@@ -61,27 +75,34 @@ func randString(n int, src rand.Source, prefix string) string {
 	return prefix + string(b[0:30-len(prefix)])
 }
 
+func isQuickMode() bool {
+	return os.Getenv("MODE") == "quick"
+}
+
 // Tests bucket re-create errors.
-func TestMakeBucketError(t *testing.T) {
-	if testing.Short() {
-		t.Skip("skipping functional tests for short runs")
+func testMakeBucketError() {
+	logTrace()
+	if isQuickMode() {
```

Comment:
We are better off doing this in main()

```
if !isQuickMode() {
   .. long running tests.
}
```
Avoids unrelated code creeping into test body code.